metacat-metadata-mysql/src/main/java/com/netflix/metacat/metadata/mysql/MySqlParentChildRelMetaDataService.java

```diff
@@ -141,8 +141,8 @@ public void drop(final QualifiedName name) {
                     return ps;
                 });
 
-                if (count == 0) {
-                    throw new RuntimeException("Fail to drop " + name.toString() + " because it still has child table");
+                if (count == 0 && getChildren(name).size() != 0) {
```

Comment:
Need to have `getChildren(name).size() != 0` because majority of the tables don ot have child table and when we try to execute the drop query, the count will always be zero. And the validation should only fail those that still have parents